code/game/objects/structures/crates_lockers/closets.dm

```diff
@@ -209,6 +210,11 @@
 	else
 		return open(user)
 
+/obj/structure/closet/proc/close_storage(mob/living/user)
+	for(var/obj/item/storage/opened in contents)
+		var/datum/component/storage/closeall = opened.GetComponent(/datum/component/storage)
+		closeall.close_all()
```

Comment:
```suggestion
	for(var/obj/object in contents)
		var/datum/component/storage/closeall = opened.GetComponent(/datum/component/storage)
		if(closeall)
			closeall.close_all()
```

Reply:
Storage is a component... the point is to not need to rely on typepath for behaviour